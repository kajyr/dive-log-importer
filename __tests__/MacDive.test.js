const { importer } = require('../src/macdive');
const { read } = require('../src/xml');
const path = require('path');

const MOCK_FILE = path.join(__dirname, '__mocks__', 'MacDive.xml');

describe('MacDive importer', () => {
  test('Basic', async () => {
    const mock = await read(MOCK_FILE, 'utf8');
    const logbook = importer(mock);
    expect(logbook).toMatchSnapshot();
  });
});
