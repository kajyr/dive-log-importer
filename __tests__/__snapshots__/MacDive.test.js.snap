// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MacDive importer Basic 1`] = `
Object {
  "dives": Array [
    Object {
      "air_used": 2409.75,
      "bottom_time": 29.5,
      "buddies": "Silvia Balbo, Matteo Carli",
      "city": "Santo Stefano al Mare",
      "computer": "Teric",
      "country": "Italy",
      "current": "Nessuna",
      "current_is_calm": true,
      "current_is_strong": false,
      "current_is_weak": false,
      "date": "2019-05-02",
      "deco_stops": "-",
      "depths": Array [
        0.8,
        2.6,
        3,
        4.6,
        7.1,
        9.5,
        11.4,
        14.4,
        17.3,
        19.3,
        22.2,
        25.4,
        28.1,
        31.5,
        32.9,
        32.8,
        32.7,
        32.7,
        32.7,
        32.6,
        32.5,
        32.3,
        32.7,
        32.6,
        32.7,
        32.9,
        32.7,
        32.5,
        32.5,
        32.3,
        32.5,
        32.6,
        32.7,
        32.4,
        32.5,
        32.2,
        32.5,
        32.6,
        32.8,
        33.4,
        33.1,
        32.7,
        32.8,
        32.8,
        32.9,
        32.6,
        33,
        32.4,
        32.6,
        32.5,
        32.7,
        32.9,
        32.1,
        32.2,
        32.4,
        32.3,
        32.5,
        32.4,
        32.3,
        32.1,
        32,
        31.6,
        31.2,
        31.2,
        31.3,
        31.2,
        31.5,
        31.7,
        31.7,
        32,
        32.2,
        32,
        31.5,
        31.4,
        31,
        31.1,
        31,
        30.6,
        30.6,
        30.3,
        29.8,
        29.7,
        29.7,
        30.1,
        30,
        30.5,
        30.1,
        30.2,
        29.9,
        29.9,
        30,
        30.4,
        30.7,
        30.8,
        31.1,
        31.4,
        31.8,
        32.3,
        32.5,
        32.6,
        32.6,
        32.8,
        32.8,
        32.7,
        32.9,
        32.9,
        32.8,
        32.9,
        32.8,
        32.6,
        32.5,
        32.3,
        31.6,
        31.4,
        31.2,
        31.3,
        31.3,
        31.3,
        31.7,
        31.5,
        31.1,
        31.2,
        31.6,
        31.5,
        31.7,
        32,
        32.2,
        32.4,
        32,
        32.1,
        31.8,
        31.5,
        31.3,
        31.4,
        31.6,
        31.4,
        30.9,
        31.4,
        31.3,
        31.1,
        30.8,
        30.8,
        31.4,
        31.7,
        32.2,
        32.5,
        32.1,
        32.5,
        32.6,
        32.4,
        32,
        32.3,
        31.5,
        31.1,
        31,
        31.6,
        31.8,
        32,
        31.2,
        30.2,
        29.3,
        28,
        26.5,
        24.6,
        23.2,
        22,
        21.4,
        21.1,
        21.5,
        20.9,
        19.8,
        19.6,
        18.5,
        17.2,
        16.9,
        16.5,
        15.5,
        14.6,
        13.1,
        13.2,
        13.6,
        12.5,
        11.2,
        9.7,
        9.5,
        8.7,
        7.4,
        5.9,
        6,
        5.2,
        5.9,
        6.2,
        6.2,
        6.4,
        6.1,
        6.6,
        6.7,
        6.2,
        5.3,
        6.4,
        7.3,
        7.2,
        7.6,
        6.9,
        6.3,
        5.9,
        6.1,
        6.6,
        6.6,
        6.4,
        5.5,
        5.3,
        4.5,
        5,
        4.9,
        5.3,
        6,
        6.4,
        6.6,
        6.4,
        5.3,
        4.8,
        5.1,
        5.8,
        5.5,
        4.7,
        3.8,
        4.5,
        4.3,
        5.6,
        5.6,
        5.1,
        3.5,
        1.5,
        0,
        0.4,
        0,
        0,
        0,
        0,
        0,
        0,
        0.4,
        0.4,
        0,
        0,
        0,
        0,
        0,
        0.4,
      ],
      "diveSuit": "Rash guard",
      "diveTime": 41,
      "dive_master": "Matteo Carli",
      "emersion_time": 4,
      "entry": "da barca",
      "entry_time": "15:31",
      "exit_time": "16:12",
      "half_depth_break": "17m",
      "half_depth_break_time": "2.5",
      "isAir": false,
      "lat": "",
      "long": "",
      "max_depth": 33.5,
      "number": 60,
      "pressure_end": 33.51,
      "pressure_start": 194.16,
      "repetitive": false,
      "site": "Isola",
      "surface": "Medio mosso",
      "surfaceInterval": "-",
      "surface_is_calm": false,
      "surface_is_mid": false,
      "surface_is_rough": false,
      "temps": Array [
        18,
        18,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        15,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        15,
        15,
        15,
        15,
        16,
        16,
        16,
        16,
        16,
        15,
        16,
        16,
        16,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        16,
        15,
        15,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        16,
        16,
        16,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        15,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
      ],
      "times": Array [
        0,
        10,
        20,
        30,
        40,
        50,
        60,
        70,
        80,
        90,
        100,
        110,
        120,
        130,
        140,
        150,
        160,
        170,
        180,
        190,
        200,
        210,
        220,
        230,
        240,
        250,
        260,
        270,
        280,
        290,
        300,
        310,
        320,
        330,
        340,
        350,
        360,
        370,
        380,
        390,
        400,
        410,
        420,
        430,
        440,
        450,
        460,
        470,
        480,
        490,
        500,
        510,
        520,
        530,
        540,
        550,
        560,
        570,
        580,
        590,
        600,
        610,
        620,
        630,
        640,
        650,
        660,
        670,
        680,
        690,
        700,
        710,
        720,
        730,
        740,
        750,
        760,
        770,
        780,
        790,
        800,
        810,
        820,
        830,
        840,
        850,
        860,
        870,
        880,
        890,
        900,
        910,
        920,
        930,
        940,
        950,
        960,
        970,
        980,
        990,
        1000,
        1010,
        1020,
        1030,
        1040,
        1050,
        1060,
        1070,
        1080,
        1090,
        1100,
        1110,
        1120,
        1130,
        1140,
        1150,
        1160,
        1170,
        1180,
        1190,
        1200,
        1210,
        1220,
        1230,
        1240,
        1250,
        1260,
        1270,
        1280,
        1290,
        1300,
        1310,
        1320,
        1330,
        1340,
        1350,
        1360,
        1370,
        1380,
        1390,
        1400,
        1410,
        1420,
        1430,
        1440,
        1450,
        1460,
        1470,
        1480,
        1490,
        1500,
        1510,
        1520,
        1530,
        1540,
        1550,
        1560,
        1570,
        1580,
        1590,
        1600,
        1610,
        1620,
        1630,
        1640,
        1650,
        1660,
        1670,
        1680,
        1690,
        1700,
        1710,
        1720,
        1730,
        1740,
        1750,
        1760,
        1770,
        1780,
        1790,
        1800,
        1810,
        1820,
        1830,
        1840,
        1850,
        1860,
        1870,
        1880,
        1890,
        1900,
        1910,
        1920,
        1930,
        1940,
        1950,
        1960,
        1970,
        1980,
        1990,
        2000,
        2010,
        2020,
        2030,
        2040,
        2050,
        2060,
        2070,
        2080,
        2090,
        2100,
        2110,
        2120,
        2130,
        2140,
        2150,
        2160,
        2170,
        2180,
        2190,
        2200,
        2210,
        2220,
        2230,
        2240,
        2250,
        2260,
        2270,
        2280,
        2290,
        2300,
        2310,
        2320,
        2330,
        2340,
        2350,
        2360,
        2370,
        2380,
        2390,
        2400,
        2410,
        2420,
        2430,
        2440,
        2450,
        2460,
        2470,
        2480,
        2490,
      ],
      "type": "",
      "visibility": "Buona",
      "visibility_is_enough": false,
      "visibility_is_good": true,
      "visibility_is_poor": false,
      "volume_start": 2912.4,
      "volume_tank": 15,
      "water": "mare",
      "weather": "Grigio",
      "weather_is_clear": false,
      "weather_is_cloud": false,
      "weather_is_rain": false,
      "weights": 8,
    },
    Object {
      "air_used": 2039.5500000000004,
      "bottom_time": 35.5,
      "buddies": "Giuseppe Costa, Valentina Tanduo",
      "city": "Isola d'Elba",
      "computer": "Teric",
      "country": "Italy",
      "current": "Nessuna",
      "current_is_calm": true,
      "current_is_strong": false,
      "current_is_weak": false,
      "date": "2019-04-27",
      "deco_stops": "-",
      "depths": Array [
        0.3,
        1.9,
        1.4,
        2.6,
        2.1,
        1.2,
        1.8,
        6.1,
        9,
        11.7,
        12.8,
        12.5,
        12.4,
        12.9,
        13.4,
        14.4,
        16.1,
        16.7,
        16.9,
        16.6,
        17.1,
        16.9,
        16.9,
        17.2,
        17.4,
        17.4,
        17.1,
        16.9,
        15.7,
        15.8,
        16,
        16.2,
        15.8,
        15.5,
        15.6,
        15.7,
        15.2,
        15.5,
        15.2,
        15.3,
        15.6,
        15.6,
        16,
        15.8,
        16,
        16.1,
        15.9,
        16.3,
        15.7,
        15.4,
        16.1,
        15.6,
        15.4,
        16,
        15.5,
        15.3,
        15.3,
        16,
        16.1,
        16,
        16.3,
        16.5,
        16.8,
        17.8,
        18.1,
        18.6,
        18.9,
        19.7,
        20.5,
        19.5,
        18.7,
        18.2,
        18.3,
        18,
        17.9,
        19,
        18.6,
        18.5,
        18.4,
        19.1,
        18.5,
        17.9,
        17.8,
        17.8,
        16.1,
        16.4,
        16.7,
        17.1,
        16.9,
        17.2,
        16.8,
        16.7,
        16.8,
        17,
        16.3,
        15.9,
        15.7,
        15.6,
        15.1,
        14.7,
        14.7,
        15.3,
        15.3,
        15.7,
        15.9,
        15.5,
        15.3,
        15.1,
        15,
        14.5,
        14.7,
        15.1,
        15.2,
        15.1,
        15,
        15.2,
        15.5,
        15.9,
        16.4,
        16.5,
        16.4,
        16.3,
        15.7,
        16.3,
        16.4,
        15.9,
        16.6,
        16.3,
        16.2,
        16.8,
        17.2,
        17.4,
        18.3,
        18.5,
        17.9,
        17.3,
        17,
        16.7,
        16.3,
        16.8,
        17.1,
        16.9,
        16.6,
        16.7,
        16.7,
        16.7,
        16.8,
        17.1,
        17,
        16.8,
        16.7,
        17.1,
        16.8,
        16.7,
        16.8,
        16.8,
        17,
        16.7,
        16.5,
        16.3,
        15.9,
        16.1,
        15.7,
        14.7,
        14.8,
        14.1,
        15.2,
        15.2,
        15.2,
        15.2,
        15.4,
        15.5,
        15.7,
        16.1,
        16,
        16,
        16.2,
        17,
        17.6,
        17.1,
        16,
        16,
        16.3,
        16.8,
        16.5,
        15.8,
        15.4,
        15.1,
        14.8,
        15.2,
        15.6,
        15.9,
        15.7,
        16.4,
        16.3,
        16.1,
        16.7,
        16.6,
        16.4,
        16.2,
        16,
        16.2,
        15.9,
        16.1,
        16.4,
        16,
        15.7,
        15.3,
        15,
        15.3,
        15.3,
        15.5,
        15.2,
        15.1,
        15.8,
        15.5,
        15.9,
        16.3,
        16.1,
        16.8,
        16.4,
        15.8,
        15.4,
        14.4,
        14.4,
        13.5,
        14.2,
        14.2,
        12.9,
        13,
        12.9,
        12,
        11.3,
        10.6,
        10.3,
        9.9,
        9,
        8.6,
        7.8,
        7,
        5.9,
        5.9,
        5.5,
        4.4,
        5.2,
        5.7,
        6.4,
        6.3,
        6.4,
        6.9,
        6.8,
        7.5,
        6.6,
        6.3,
        5.2,
        4.4,
        4.6,
        4.4,
        4.7,
        4.3,
        3.7,
        2.2,
        1.6,
        1.5,
        0.6,
        0,
        0.4,
        0.5,
        0.4,
        0.3,
        0.6,
        0.5,
        0.5,
        0.6,
        0.7,
      ],
      "diveSuit": "Rash guard",
      "diveTime": 45,
      "dive_master": "Valentina Tanduo",
      "emersion_time": 2,
      "entry": "Da Barca",
      "entry_time": "14:21",
      "exit_time": "15:06",
      "half_depth_break": "11m",
      "half_depth_break_time": "2.5",
      "isAir": true,
      "lat": "",
      "long": "",
      "max_depth": 20.7,
      "number": 59,
      "pressure_end": 54.33,
      "pressure_start": 190.3,
      "repetitive": true,
      "site": "Scoglio della Nave",
      "surface": "Calmo",
      "surfaceInterval": "3:44",
      "surface_is_calm": true,
      "surface_is_mid": false,
      "surface_is_rough": false,
      "temps": Array [
        18,
        18,
        18,
        18,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
      ],
      "times": Array [
        0,
        10,
        20,
        30,
        40,
        50,
        60,
        70,
        80,
        90,
        100,
        110,
        120,
        130,
        140,
        150,
        160,
        170,
        180,
        190,
        200,
        210,
        220,
        230,
        240,
        250,
        260,
        270,
        280,
        290,
        300,
        310,
        320,
        330,
        340,
        350,
        360,
        370,
        380,
        390,
        400,
        410,
        420,
        430,
        440,
        450,
        460,
        470,
        480,
        490,
        500,
        510,
        520,
        530,
        540,
        550,
        560,
        570,
        580,
        590,
        600,
        610,
        620,
        630,
        640,
        650,
        660,
        670,
        680,
        690,
        700,
        710,
        720,
        730,
        740,
        750,
        760,
        770,
        780,
        790,
        800,
        810,
        820,
        830,
        840,
        850,
        860,
        870,
        880,
        890,
        900,
        910,
        920,
        930,
        940,
        950,
        960,
        970,
        980,
        990,
        1000,
        1010,
        1020,
        1030,
        1040,
        1050,
        1060,
        1070,
        1080,
        1090,
        1100,
        1110,
        1120,
        1130,
        1140,
        1150,
        1160,
        1170,
        1180,
        1190,
        1200,
        1210,
        1220,
        1230,
        1240,
        1250,
        1260,
        1270,
        1280,
        1290,
        1300,
        1310,
        1320,
        1330,
        1340,
        1350,
        1360,
        1370,
        1380,
        1390,
        1400,
        1410,
        1420,
        1430,
        1440,
        1450,
        1460,
        1470,
        1480,
        1490,
        1500,
        1510,
        1520,
        1530,
        1540,
        1550,
        1560,
        1570,
        1580,
        1590,
        1600,
        1610,
        1620,
        1630,
        1640,
        1650,
        1660,
        1670,
        1680,
        1690,
        1700,
        1710,
        1720,
        1730,
        1740,
        1750,
        1760,
        1770,
        1780,
        1790,
        1800,
        1810,
        1820,
        1830,
        1840,
        1850,
        1860,
        1870,
        1880,
        1890,
        1900,
        1910,
        1920,
        1930,
        1940,
        1950,
        1960,
        1970,
        1980,
        1990,
        2000,
        2010,
        2020,
        2030,
        2040,
        2050,
        2060,
        2070,
        2080,
        2090,
        2100,
        2110,
        2120,
        2130,
        2140,
        2150,
        2160,
        2170,
        2180,
        2190,
        2200,
        2210,
        2220,
        2230,
        2240,
        2250,
        2260,
        2270,
        2280,
        2290,
        2300,
        2310,
        2320,
        2330,
        2340,
        2350,
        2360,
        2370,
        2380,
        2390,
        2400,
        2410,
        2420,
        2430,
        2440,
        2450,
        2460,
        2470,
        2480,
        2490,
        2500,
        2510,
        2520,
        2530,
        2540,
        2550,
        2560,
        2570,
        2580,
        2590,
        2600,
        2610,
        2620,
        2630,
        2640,
        2650,
        2660,
        2670,
        2680,
        2690,
        2700,
        2710,
        2720,
        2730,
        2740,
      ],
      "type": "corso 1° grado",
      "visibility": "Ottima",
      "visibility_is_enough": false,
      "visibility_is_good": false,
      "visibility_is_poor": false,
      "volume_start": 2854.5,
      "volume_tank": 15,
      "water": "mare",
      "weather": "Sole",
      "weather_is_clear": true,
      "weather_is_cloud": false,
      "weather_is_rain": false,
      "weights": 8,
    },
    Object {
      "air_used": 1547.1,
      "bottom_time": 29.5,
      "buddies": "Valentina Tanduo",
      "city": "Isola d'Elba",
      "computer": "Teric",
      "country": "Italy",
      "current": "Nessuna",
      "current_is_calm": true,
      "current_is_strong": false,
      "current_is_weak": false,
      "date": "2019-04-27",
      "deco_stops": "-",
      "depths": Array [
        0.7,
        2.7,
        3.5,
        4.2,
        4.2,
        4.2,
        4,
        3.7,
        2.9,
        1.5,
        3.5,
        4.5,
        6,
        6,
        6.3,
        6.2,
        6.2,
        6,
        6.8,
        7.8,
        9,
        9.4,
        9.6,
        9,
        9.5,
        10,
        9.8,
        9.9,
        9.9,
        10.3,
        10.7,
        11,
        11.8,
        12.2,
        12.3,
        12.7,
        13.3,
        13.8,
        13.3,
        13,
        12.7,
        12.8,
        14,
        13.7,
        13.7,
        13.9,
        14,
        14.6,
        14.6,
        14.7,
        14.5,
        14.9,
        14.7,
        14.7,
        15,
        15.4,
        15.4,
        15.4,
        15.3,
        15,
        15.1,
        15.7,
        16.3,
        16.8,
        16.6,
        16.7,
        16.8,
        17,
        17.2,
        17.1,
        17,
        17,
        16.5,
        16,
        15.9,
        15.4,
        15.7,
        16.1,
        16.8,
        17.2,
        17.4,
        17.1,
        17.5,
        17.9,
        17.9,
        18.4,
        18.5,
        18.9,
        19.2,
        20.1,
        20.9,
        21.5,
        21.1,
        21,
        20.8,
        21.4,
        21.3,
        21.2,
        20.9,
        19.9,
        19.4,
        18.4,
        16.9,
        17,
        17.7,
        18,
        18.2,
        18.4,
        18.7,
        18,
        17.8,
        16.9,
        16.6,
        16.3,
        15.4,
        15.6,
        16,
        17.2,
        16.8,
        15.5,
        14.4,
        13.9,
        13.6,
        13.6,
        14.3,
        13.4,
        12.4,
        12.6,
        13.2,
        11,
        11.1,
        12.4,
        12.5,
        12.2,
        11.7,
        11.6,
        11.7,
        12.3,
        13.5,
        14.9,
        16.1,
        16.3,
        16.3,
        15.8,
        15.1,
        14.7,
        14.7,
        14.9,
        15,
        14.5,
        13.9,
        13.5,
        12.7,
        12,
        11.5,
        11.2,
        10.8,
        10.9,
        11.1,
        11.5,
        12.1,
        12.2,
        13,
        13.7,
        13.6,
        13.7,
        13.6,
        13.7,
        13.8,
        13.3,
        13.2,
        13.1,
        12.7,
        12.3,
        11.9,
        12.1,
        11.7,
        11.9,
        11.3,
        11.4,
        11.2,
        11,
        10.7,
        11,
        10.7,
        11,
        11.1,
        11.1,
        11.5,
        11.9,
        13.4,
        13.1,
        12.7,
        12.3,
        12.1,
        10.6,
        9.5,
        8.8,
        8.1,
        7.9,
        6.5,
        5.8,
        5.6,
        5.4,
        5.4,
        5.8,
        5.4,
        5.3,
        5.8,
        6.1,
        5.3,
        5.1,
        4.7,
        5.2,
        6.2,
        6,
        6.5,
        6.6,
        6.6,
        6.2,
        5.8,
        5.4,
        4.6,
        3.8,
        2.9,
        2.3,
        1.6,
        0.9,
        0,
        0,
        0.3,
        0,
        0,
        0.3,
        0,
        0.5,
        0.4,
        0.5,
      ],
      "diveSuit": "Rash guard",
      "diveTime": 39,
      "dive_master": "Valentina Tanduo",
      "emersion_time": 2,
      "entry": "Da Barca",
      "entry_time": "10:00",
      "exit_time": "10:39",
      "half_depth_break": "11m",
      "half_depth_break_time": "2.5",
      "isAir": true,
      "lat": "",
      "long": "",
      "max_depth": 21.6,
      "number": 58,
      "pressure_end": 103.7,
      "pressure_start": 206.84,
      "repetitive": true,
      "site": "Punta del nasuto",
      "surface": "Calmo",
      "surfaceInterval": "17:21",
      "surface_is_calm": true,
      "surface_is_mid": false,
      "surface_is_rough": false,
      "temps": Array [
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        17,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
        16,
      ],
      "times": Array [
        0,
        10,
        20,
        30,
        40,
        50,
        60,
        70,
        80,
        90,
        100,
        110,
        120,
        130,
        140,
        150,
        160,
        170,
        180,
        190,
        200,
        210,
        220,
        230,
        240,
        250,
        260,
        270,
        280,
        290,
        300,
        310,
        320,
        330,
        340,
        350,
        360,
        370,
        380,
        390,
        400,
        410,
        420,
        430,
        440,
        450,
        460,
        470,
        480,
        490,
        500,
        510,
        520,
        530,
        540,
        550,
        560,
        570,
        580,
        590,
        600,
        610,
        620,
        630,
        640,
        650,
        660,
        670,
        680,
        690,
        700,
        710,
        720,
        730,
        740,
        750,
        760,
        770,
        780,
        790,
        800,
        810,
        820,
        830,
        840,
        850,
        860,
        870,
        880,
        890,
        900,
        910,
        920,
        930,
        940,
        950,
        960,
        970,
        980,
        990,
        1000,
        1010,
        1020,
        1030,
        1040,
        1050,
        1060,
        1070,
        1080,
        1090,
        1100,
        1110,
        1120,
        1130,
        1140,
        1150,
        1160,
        1170,
        1180,
        1190,
        1200,
        1210,
        1220,
        1230,
        1240,
        1250,
        1260,
        1270,
        1280,
        1290,
        1300,
        1310,
        1320,
        1330,
        1340,
        1350,
        1360,
        1370,
        1380,
        1390,
        1400,
        1410,
        1420,
        1430,
        1440,
        1450,
        1460,
        1470,
        1480,
        1490,
        1500,
        1510,
        1520,
        1530,
        1540,
        1550,
        1560,
        1570,
        1580,
        1590,
        1600,
        1610,
        1620,
        1630,
        1640,
        1650,
        1660,
        1670,
        1680,
        1690,
        1700,
        1710,
        1720,
        1730,
        1740,
        1750,
        1760,
        1770,
        1780,
        1790,
        1800,
        1810,
        1820,
        1830,
        1840,
        1850,
        1860,
        1870,
        1880,
        1890,
        1900,
        1910,
        1920,
        1930,
        1940,
        1950,
        1960,
        1970,
        1980,
        1990,
        2000,
        2010,
        2020,
        2030,
        2040,
        2050,
        2060,
        2070,
        2080,
        2090,
        2100,
        2110,
        2120,
        2130,
        2140,
        2150,
        2160,
        2170,
        2180,
        2190,
        2200,
        2210,
        2220,
        2230,
        2240,
        2250,
        2260,
        2270,
        2280,
        2290,
        2300,
        2310,
        2320,
        2330,
        2340,
        2350,
        2360,
        2370,
      ],
      "type": "corso 1° grado",
      "visibility": "Ottima",
      "visibility_is_enough": false,
      "visibility_is_good": false,
      "visibility_is_poor": false,
      "volume_start": 3102.6,
      "volume_tank": 15,
      "water": "mare",
      "weather": "Sole",
      "weather_is_clear": true,
      "weather_is_cloud": false,
      "weather_is_rain": false,
      "weights": 8,
    },
  ],
}
`;
